js/deer-record.js

```diff
@@ -263,86 +263,93 @@ export default class DeerReport {
             }
             let flatKeys = annotations.map(input => input.getAttribute(DEER.KEY))
             annotations = annotations.filter((el, i) => {
-                    //Throw a soft error if we detect duplicate deer-key entries, and only respect the first one.
-                    if (flatKeys.indexOf(el.getAttribute(DEER.KEY)) !== i) {
-                        UTILS.warning("Duplicate input " + DEER.KEY + " attribute value '" + el.getAttribute(DEER.KEY) + "' detected during submission.  This input will be ignored.  See duplicate below. ", el)
-                    }
-                    return flatKeys.indexOf(el.getAttribute(DEER.KEY)) === i
-                })
-                .map(input => {
-                    let inputId = input.getAttribute(DEER.SOURCE)
-                    let action = (inputId) ? "UPDATE" : "CREATE"
-                    let annotation = {
-                        type: "Annotation",
-                        creator: DEER.ATTRIBUTION,
-                        target: entity["@id"],
-                        body: {}
+                //Throw a soft error if we detect duplicate deer-key entries, and only respect the first one.
+                if (flatKeys.indexOf(el.getAttribute(DEER.KEY)) !== i) {
+                    UTILS.warning("Duplicate input " + DEER.KEY + " attribute value '" + el.getAttribute(DEER.KEY) + "' detected during submission.  This input will be ignored.  See duplicate below. ", el)
+                }
+                return flatKeys.indexOf(el.getAttribute(DEER.KEY)) === i
+            })
+            .filter(el => {
+                //If this is a DEER.PRIMITIVES entry DO NOT repeat it as an annotation.  
+                if (DEER.PRIMITIVES.includes(el.getAttribute(DEER.KEY))) {
+                    UTILS.warning("A DEER.PRIMITIVES entry was detected (" + el.getAttribute(DEER.KEY) + ").  An annotation will not be created for this entry since it is directly on the entity.")
```

Comment:
This is not smart enough. We need the use case where a primitive is updated.

```json
 { "@type": "Thing",
   "primitive": "A" }
```

should be allowed to update so that the expanded form is `"primitive":"B"` without having to update the original object. An updating annotation will do this and should not be filtered out.

Reply:
Then that needs to be handled in `processRecord()` as an entity updater before it gets to the annotations.  By this point, all primitive handling is expected to have taken place, which is why it can be ignored. 

It needs to be handled just above this code around
```javascript
        if (this.id) {
            /* RIGHT HERE I THINK.  Has a primitive changed?  Then we need to update the entity. */
            record["@id"] = this.id
            formAction = Promise.resolve(record)
        } else {
            let self = this
            formAction = fetch(DEER.URLS.CREATE, {
                method: "POST",
                headers: {
                    "Content-Type": "application/json; charset=utf-8"
                },
                body: JSON.stringify(record)
            })
            .then(response => response.json())
            .then(data =>{
                UTILS.broadcast(undefined, DEER.EVENTS.CREATED, self.elem, data.new_obj_state)
                return data.new_obj_state
            })
            .catch(err => {})
        }

      /* Now handle the annotations in formAction.then().  Primitives do not need to be considered.  */
```